odoo/addons/base/models/ir_ui_view.py

```diff
@@ -581,62 +584,84 @@ def _get_filter_xmlid_query(self):
                   WHERE res_id IN %(res_ids)s AND model = 'ir.ui.view' AND module IN %(modules)s
                """
 
-    def get_inheriting_views_arch(self, model):
-        """Retrieves the sets of views that should currently be used in the
-           system in the right order. During the module upgrade phase it
-           may happen that a view is present in the database but the fields it relies on are not
-           fully loaded yet. This method only considers views that belong to modules whose code
-           is already loaded. Custom views defined directly in the database are loaded only
-           after the module initialization phase is completely finished.
-
-           :param str model: model identifier of the inheriting views.
-           :return: list of ir.ui.view
+    def _get_inheriting_views(self):
         """
-        self.ensure_one()
-        self.check_access_rights('read')
```

Comment:
I believe we need to keep this check even if `_get_inheriting_views()` is private, as it is reachable from other public methods and the raw SQL is not otherwise protected.

e.g. I can now call `get_combined_arch()` in RPC without being an admin, and I get the view arch